text/0000-private_data_purge.md

```diff
@@ -0,0 +1,179 @@
+
+### New Keys in Private data store
+ In this design, we maintain two types of additional keys in the private data store. We refer to these new types of keys as *PurgeMarker* and *HashedIndex*. Below, is the format and intended use of these keys.
+1) PurgeMarker
+   A PurgeMarker key is intended to record a purge operation for a key. The design of this marker key would be `Ns_Coll_Hash(PrivateKey)_CommittingBlkNum_CommittingTxNum`. For illustration, assume that a purge transaction appears in block number 100, at position 20 that specifies for purging the keys `key1` and `key2` under collection `myColl` for chaincode `myCC`. When processing this transaction, we insert the two keys into private data store as `myCC_myColl_hash(key1)_100_20` and `myCC_myColl_hash(Key2)_100_20`. Note that the separation of different components of a key shown by an understore is just for the illustration. For the reasons discussed latter, we intend to keep these marker keys forever. However, the assumption is that the purge operation is expected to be relatively less frequent, and hence, it should not be a concern from storage space point of view.
+
+2) HashedIndex
+   A HashedIndex key is intended to act as an index for the data keys in private data store, so that when a key is eventually purged we know from where to purge it in the private data store. The format of this key would be `Ns_Coll_Hash(PrivateKey)_blkNum_txNum`. For each version of a private key in the private data store, we would maintain such an index key. For illustration, assume that a key `key1` under collection `myColl` in chaincode `myCC` is inserted at block 100, transaction 10. Later this key is updated in block 200, transaction 20 and finally, the key is deleted in block 300, transaction 30. In this case, today, we maintain in the private data store all three private write-sets that contain, potentially along with other, the above three mentioned operations. In this example, for `key1`, we would maintain three HashedIndex keys as `myCC_myColl_hash(key1)_100_10`, `myCC_myColl_hash(key1)_200_20`, and `myCC_myColl_hash(key1)_300_30`. The value for the HashedIndex key will be the private key itself. For instance, in the above two examples, the values would be `key1` and `key2` respectively.
+   
+   - Note that the format of this key look same as the format of the PurgeMarker key - however, we always prepend a unique byte for the key type so, no two keys of differnt type ever clash.
+   
+   - Unlike the PurgeMarker key, we delete an HashedIndex key when we delete the corresponding data, either because of newly introduce explicit purge operation in this RFC or because of an implicit purge operation caused by [expiry of private data](https://hyperledger-fabric.readthedocs.io/en/release-2.2/private_data_tutorial.html#purge-private-data).
```

Comment:
```suggestion
   - Unlike the PurgeMarker key, we delete an HashedIndex key when we delete the corresponding data, either because of newly introduced explicit purge operation in this RFC or because of an implicit purge operation caused by [expiry of private data](https://hyperledger-fabric.readthedocs.io/en/release-2.2/private_data_tutorial.html#purge-private-data).
```